pandas/tools/merge.py

```diff
@@ -896,8 +896,10 @@ def get_result(self):
                 data = dict(zip(range(len(self.objs)), self.objs))
                 index, columns = self.new_axes
                 tmpdf = DataFrame(data, index=index)
-                if columns is not None:
-                    tmpdf.columns = columns
+                columns = []
+                for i in range(len(data)):
+                    columns.append(Series(data[i]).name if Series(data[i]).name is not None else i)
+                tmpdf.columns = columns
```

Comment:
Hi jreback. I see your point. I've erroneously believed we wanted to use the column index as the column name in case the parent Series didn't have a valid name. I'm going to update the pull request to include the change you suggested. 

Apologies in advance if my code is not always the neatest. I'm new to software development, although I use pandas everyday for data analysis. I thought actively participating to the project could be a good way for me to learn how to code properly and become a better programmer.